src/OpenTelemetry/Metrics/Processors/MetricProcessor.cs

```diff
@@ -14,9 +14,14 @@
 // limitations under the License.
 // </copyright>
 
+using System;
+
 namespace OpenTelemetry.Metrics
 {
     public abstract class MetricProcessor : BaseProcessor<MetricItem>
     {
+        // GetMetric or GetMemoryState or GetAggregatedMetrics..
+        // ...or some other names
+        public abstract void SetGetMetricFunction(Func<MetricItem> getMetrics);
```

Comment:
I might not be thinking about things right, but would it make sense for a MetricProcessor to have a handle on the ParentProvider kinda like how we do for traces? That way instead of calling `SetGetMetricFunction(this.Collect)` in the MeterProviderSdk you'd have a handle on the provider to call `Collect` directly.

Reply:
I like this idea. Will explore more on and try it in the subsequent PRs.